python/rapidsmpf/rapidsmpf/integrations/cudf/partition.pyx

```diff
@@ -0,0 +1,223 @@
+    Table table,
+    columns_to_hash,
+    int num_partitions,
+    stream,
+    DeviceMemoryResource device_mr,
+):
+    """
+    Partition rows from the input table into multiple packed (serialized) tables.
+
+    Parameters
+    ----------
+    table
+        The input table to partition.
+    columns_to_hash
+        Indices of the input columns to use for hashing.
+    num_partitions
+        The number of partitions to create.
+    stream
+        The CUDA stream used for memory operations.
+    device_mr
+        Reference to the RMM device memory resource used for device allocations.
+
+    Returns
+    -------
+    A dictionary where the keys are partition IDs and the values are packed tables.
+
+    Raises
+    ------
+    IndexError
+        If an index in ``columns_to_hash`` is invalid.
+
+    See Also
+    --------
+    rapidsmpf.integrations.cudf.partition.unpack_and_concat
+    pylibcudf.partitioning.hash_partition
+    pylibcudf.contiguous_split.pack
+    rapidsmpf.integrations.cudf.partition.split_and_pack
+    """
+    cdef vector[size_type] _columns_to_hash = tuple(columns_to_hash)
+    cdef unordered_map[uint32_t, cpp_PackedData] _ret
+    cdef table_view tbl = table.view()
+    if stream is None:
+        raise ValueError("stream cannot be None")
+    cdef cuda_stream_view _stream = Stream(stream).view()
+    with nogil:
+        _ret = cpp_partition_and_pack(
+            tbl,
+            _columns_to_hash,
+            num_partitions,
+            cpp_HASH_MURMUR3,
+            cpp_DEFAULT_HASH_SEED,
+            _stream,
+            device_mr.get_mr()
+        )
+    ret = {}
+    cdef unordered_map[uint32_t, cpp_PackedData].iterator it = _ret.begin()
+    while(it != _ret.end()):
+        ret[deref(it).first] = PackedData.from_librapidsmpf(
+            make_unique[cpp_PackedData](move(deref(it).second))
+        )
+        postincrement(it)
+    return ret
+
+
+cpdef dict split_and_pack(
+    Table table,
+    splits,
+    stream,
+    DeviceMemoryResource device_mr,
+):
+    """
+    Splits rows from the input table into multiple packed (serialized) tables.
+
+    Parameters
+    ----------
+    table
+        The input table to split and pack.  The table cannot be empty (the
+        split points would not be valid).
+    splits
+        The split points, equivalent to cudf::split(), i.e. one less than
+        the number of result partitions.
+    stream
+        The CUDA stream used for memory operations.
+    device_mr
+        Reference to the RMM device memory resource used for device allocations.
+
+    Returns
+    -------
+    A dictionary where the keys are partition IDs and the values are packed tables.
+
+    Raises
+    ------
+    IndexError
+        If the splits are out of range for ``[0, len(table)]``.
+
+    See Also
+    --------
+    rapidsmpf.integrations.cudf.partition.unpack_and_concat
+    pylibcudf.copying.split
+    rapidsmpf.integrations.cudf.partition_and_pack
```

Comment:
I think this isn't available here currently. We'll need to add it to `__init__.py` (or `.pyx?`) or update the reference to include this module, `.partition`:

```suggestion
    rapidsmpf.integrations.cudf.partition.partition_and_pack
```

Same for L206.